### ISSUE TITLE
elevation downloader can fail due to "slow down"

### ISSUE THREAD

**Issue by nilsnolde** [valhalla/valhalla#3390]
I'm just monitoring a first planet build with the recent changes to the elevation downloader. 3 tiles couldn't be downloaded caused by what I assume some sort of rate limit:
```
2021-11-07 17:07:46,482 CRITICAL: Download failed of elevation tile S33/S33E122.hgt: Slow Down
2021-11-07 17:07:50,412 CRITICAL: Download failed of elevation tile S20/S20E124.hgt: Slow Down
2021-11-07 17:07:51,716 CRITICAL: Download failed of elevation tile N22/N22W082.hgt: Slow Down
``` 
that comes from this section: https://github.com/valhalla/valhalla/blob/1d3133da992e161fcce37fd932d9c3416a7ecae5/scripts/valhalla_build_elevation#L201-L213

after we join the threads we should probably re-try to download all the ones which failed with this error message? there could be other errors where we don't want to re-try (such as lost connection) and it's hard to reproduce of course so for now I'd have to go by the `"if "Slow Down" in e.reason:`. or simply put a tiny sleep after each download?

**Comment by nilsnolde** [comment]
on a good note: downloading only the skadi tiles which cover a portion of the graph reduced the full elevation dataset to ~380 GB (from ~1.5 TB before).

**Comment by danpat** [comment]
@nilsnolde There are a couple of good articles that AWS has put out on how to write a well-behaving client in a system where there is a moderate probability of failure:

https://aws.amazon.com/builders-library/timeouts-retries-and-backoff-with-jitter/

You probably want slightly different strategies depending on the class of error (i.e. a HTTP 429 may include headers that indicate when it's appropriate to retry, whereas a 500-class or client-side network timeout you may just have to guess).

**Comment by nilsnolde** [comment]
yeah, I was using those strategies before too, and if we want to auto-handle all those different errors we'd have to use it here as well. I'd do smth slimmer here since:
- 3 tiles failed downloading due to a likely 429, out of ~20k. re-trying such a low number of tiles is a fine "overload" IMO
- that's kind of the only error I think should be handled. it's very likely you'll get a few of those and that one retry at the very end will fix that. with network issues you likely have them for a longer while than a second or so. after the program failed, you can just restart it, it won't load tiles it previously downloaded so you'd only get the missing ones.
